crates/authentication/src/sys/linux/mod.rs

```diff
@@ -0,0 +1,169 @@
+    // Get authority proxy, and it's cached for future use.
+    let auth = get_authority_proxy()?;
+
+    let pid = std::process::id() as u32;
+    let subject = Subject::new_for_owner(pid, None, None)
+        .map_err(|_| Error::Unavailable)?;
+
+    // If details is non-empty then the request will fail with POLKIT_ERROR_FAILED unless the process doing the check itsef is sufficiently authorized (e.g. running as uid 0).
+    let result = auth
+        .check_authorization(
+            &subject,
+            action_id,
+            &HashMap::new(), 
+            CheckAuthorizationFlags::AllowUserInteraction.into(),
+            "",
+        )
+        .map_err(|e| map_polkit_error(e.to_string()))?;
+
+    map_authorization_result(result)
+}
+
+fn map_authorization_result(result: AuthorizationResult) -> Result<()> {
+    if result.is_authorized {
+        return Ok(());
+    }
+
+    if result.details.keys().any(|k| k.contains("dismissed")) {
+        return Err(Error::UserCanceled);
+    }
+
+    if result.is_challenge {
+        // No agent available or UI interaction not possible even though we requested it.
+        return Err(Error::Unavailable);
+    }
+
+    Err(Error::Authentication)
+}
+
+fn map_polkit_error(msg: String) -> Error {
+    let m = msg.to_lowercase();
+    if m.contains("cancel") || m.contains("canceled") || m.contains("cancelled") {
+        Error::UserCanceled
+    } else if m.contains("locked") || m.contains("exhaust") || m.contains("too many") {
+        Error::Exhausted
+    } else if m.contains("unavailable") || m.contains("no agent") || m.contains("not supported") {
+        Error::Unavailable
+    } else {
+        Error::Authentication
+    }
+}
+
+/// Authentication policy on Linux.
+/// Only action id matters.
+#[derive(Debug, Clone)]
+pub struct Policy {
+    pub(crate) action_id: &'static str,
+}
+
+/// Policy builder for Linux.
+/// On Linux, a polkit action id MUST be explicitly provided.
+/// If not set, build() returns None.
+#[derive(Debug, Clone)]
+pub struct PolicyBuilder {
+    action_id: Option<&'static str>,
+}
+
+impl PolicyBuilder {
+    #[inline]
+    pub const fn new() -> Self {
+        Self { action_id: None }
+    }
+
+    /// Required on Linux: caller must provide a polkit action id.
+    #[inline]
+    pub const fn action_id(self, id: &'static str) -> Self {
+        Self { action_id: Some(id) }
+    }
+
+    // The following are no-ops on Linux but kept for cross-platform API.
+    #[inline]
+    pub const fn biometrics(self, _strength: Option<crate::BiometricStrength>) -> Self { self }
+    #[inline]
+    pub const fn password(self, _password: bool) -> Self { self }
+    #[inline]
+    pub const fn companion(self, _companion: bool) -> Self { self }
+    #[inline]
+    pub const fn wrist_detection(self, _wrist: bool) -> Self { self }
+
+    /// Cross-platform API requirement: return Option.
+    /// Linux behavior: None if action_id is not explicitly set.
```

Comment:
this comment also needs to be moved to the top-level `build()` function, or perhaps it's enough to just have the other doc comments on the `PolicyBuilder` type itself.